puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-test.yml

```diff
@@ -0,0 +1,102 @@
+---
+- job-template:
+    name: foreman-{version}-{distro}-{action}-test
+    project-type: 'pipeline'
+    sandbox: true
+    parameters:
+      - string:
+          name: expected_version
+          default: '2.2.0-develop'
```

Comment:
I've been wondering if we could somehow make this "automatic" to avoid forgetting to change that during branching, but Jenkins has no knowledge what the "next" version will be. So I'm OK with leaving this for now.

Reply:
There's still the issue that this is hard to do with Katello. When releasing Katello 3.15.z, do we know if the Foreman version is 2.0.0 or 2.0.1 at that point? Do we care, or can we use 2.0.*? In the end it's grep so that may work.

Reply:
This entry here is only used for the Foreman pipelines. The Katello job definitions don't include this parameter. So we can align on similar pattern latter on but I believe this will not affect Katello.

Reply:
Ah right. I expected a single template that was used for Foreman, Katello and Luna. Given they all use the exact same DSL and only use different parameters.

Reply:
That would be fully true *if* I change the name of the foreman jobs to be (https://github.com/theforeman/foreman-infra/pull/1355#issuecomment-623159701), e.g:

foreman-foreman-nightly-centos7-install-test

As a reminder, we have to include `foreman-` at the beginning of job names to namespace the jobs within ci.centos.org, what follows after is our convention. I know `foreman-foreman` looks weird but if we follow the same convention for all it does make it easier to have a single definition.

Reply:
I'm happy with foreman-foreman, aligns well with our ansible stuff

Reply:
Then I'd suggest to consider `foreman-pipeline-{type}-{version}-{os}-{pipeline}`

Reply:
Another spot it helps, the function I added (https://github.com/theforeman/foreman-infra/pull/1355/files#diff-06c5ebd36e8d5d56d40a13f779b10235R100) will require modified logic to handle Foreman pipelines without strictly sticking to the convention.

Reply:
To ensure clarity, the job names would be:

foreman-pipeline-foreman-nightly-centos7-upgrade
foreman-pipeline-foreman-nightly-centos7-install
foreman-pipeline-katello-nightly-centos7-upgrade
foreman-pipeline-katello-nightly-centos8-install
foreman-pipeline-luna-nightly-centos8-install